test/unit/server_async_transaction.test.js

```diff
@@ -0,0 +1,122 @@
+const { Horizon } = StellarSdk;
+
+describe("server.js async transaction submission tests", function () {
+    let keypair = StellarSdk.Keypair.random();
+    let account = new StellarSdk.Account(keypair.publicKey(), "56199647068161");
+
+    beforeEach(function () {
+        this.server = new Horizon.Server("https://horizon-live.stellar.org:1337");
+        this.axiosMock = sinon.mock(Horizon.AxiosClient);
+        let transaction = new StellarSdk.TransactionBuilder(account, {
+            fee: 100,
+            networkPassphrase: StellarSdk.Networks.TESTNET,
+            v1: true,
+        })
+            .addOperation(
+                StellarSdk.Operation.payment({
+                    destination:
+                        "GASOCNHNNLYFNMDJYQ3XFMI7BYHIOCFW3GJEOWRPEGK2TDPGTG2E5EDW",
+                    asset: StellarSdk.Asset.native(),
+                    amount: "100.50",
+                }),
+            )
+            .setTimeout(StellarSdk.TimeoutInfinite)
+            .build();
+        transaction.sign(keypair);
+
+        this.transaction = transaction;
+        this.blob = encodeURIComponent(
+            transaction.toEnvelope().toXDR().toString("base64"),
+        );
+    });
+
+    afterEach(function () {
+        this.axiosMock.verify();
+        this.axiosMock.restore();
+    });
+
+    it("sends an async transaction", function (done) {
+        this.axiosMock
+            .expects("post")
+            .withArgs(
+                "https://horizon-live.stellar.org:1337/transactions_async",
+                `tx=${this.blob}`,
+            )
+            .returns(Promise.resolve({ data: {} }));
+
+        this.server
+            .submitAsyncTransaction(this.transaction, { skipMemoRequiredCheck: true })
+            .then(function () {
+                done();
+            })
+            .catch(function (err) {
+                done(err);
+            });
```

Comment:
suggestion: slightly more compact
```suggestion
            .then(() => done())
            .catch((err) => done(err));
```